src/api/app/models/user_ldap_strategy.rb

```diff
@@ -591,4 +582,18 @@ def self.try_ldap_con(server, user_name, password)
     Rails.logger.debug("Bound as #{user_name}")
     conn
   end
+
+  private
+
+  def relationship_groups_contains_user?(relationships, method_name)
+    relationships.each do |relationship|
+      return false if relationship.group.nil?
+      # check whether current user is in this group
+      return true if user_in_group_ldap?(login, relationship.group)
+    end
+
+    Rails.logger.info "Failed with #{method_name}"
```

Comment:
To me the current code is more straight forward / easier to understand. I would prefer to keep it this way, or to remove the method name reference. We probably don't need it.

But we can still change this later.